### PR TITLE
enable automatic release with vanniktech plugin

### PR DESCRIPTION
**Pull request by kyeotic**

### CODE REVIEW

File: gradle-mvn-publish.gradle.old
Comment (swankjesse) [inline gradle-mvn-publish.gradle.old:1]: Delete?

File: tempest-internal/build.gradle.kts [hunk 1]
```diff
@@ -1,6 +1,11 @@
+import com.vanniktech.maven.publish.JavadocJar.Dokka
+import com.vanniktech.maven.publish.MavenPublishBaseExtension
+import com.vanniktech.maven.publish.KotlinMultiplatform
+
 plugins {
   kotlin("jvm")
   `java-library`
```
Comment (swankjesse) [inline tempest-internal/build.gradle.kts:7]: Delete? (you’ve deleted `java-library` elsewhere; seems unnecessary)

File: tempest-testing-docker/build.gradle.kts [hunk 2]
```diff
@@ -1,6 +1,11 @@
+import com.vanniktech.maven.publish.JavadocJar.Dokka
+import com.vanniktech.maven.publish.MavenPublishBaseExtension
+import com.vanniktech.maven.publish.KotlinMultiplatform
+
 plugins {
   kotlin("jvm")
   `java-library`
```
Comment (swankjesse) [inline tempest-testing-docker/build.gradle.kts:7]: Same

File: tempest-testing-internal/build.gradle.kts [hunk 3]
```diff
@@ -1,6 +1,11 @@
+import com.vanniktech.maven.publish.JavadocJar.Dokka
+import com.vanniktech.maven.publish.MavenPublishBaseExtension
+import com.vanniktech.maven.publish.KotlinMultiplatform
+
 plugins {
   kotlin("jvm")
   `java-library`
```
Comment (swankjesse) [inline tempest-testing-internal/build.gradle.kts:7]: Same

File: tempest-testing-junit4/build.gradle.kts [hunk 4]
```diff
@@ -1,6 +1,11 @@
+import com.vanniktech.maven.publish.JavadocJar.Dokka
+import com.vanniktech.maven.publish.MavenPublishBaseExtension
+import com.vanniktech.maven.publish.KotlinMultiplatform
+
 plugins {
   kotlin("jvm")
   `java-library`
```
Comment (swankjesse) [inline tempest-testing-junit4/build.gradle.kts:7]: Same

File: tempest-testing-junit5/build.gradle.kts [hunk 5]
```diff
@@ -1,6 +1,11 @@
+import com.vanniktech.maven.publish.JavadocJar.Dokka
+import com.vanniktech.maven.publish.MavenPublishBaseExtension
+import com.vanniktech.maven.publish.KotlinMultiplatform
+
 plugins {
   kotlin("jvm")
   `java-library`
```
Comment (swankjesse) [inline tempest-testing-junit5/build.gradle.kts:7]: Same

File: tempest-testing-jvm/build.gradle.kts [hunk 6]
```diff
@@ -1,6 +1,11 @@
+import com.vanniktech.maven.publish.JavadocJar.Dokka
+import com.vanniktech.maven.publish.MavenPublishBaseExtension
+import com.vanniktech.maven.publish.KotlinMultiplatform
+
 plugins {
     kotlin("jvm")
     `java-library`
```
Comment (swankjesse) [inline tempest-testing-jvm/build.gradle.kts:7]: Same

File: tempest-testing/build.gradle.kts [hunk 7]
```diff
@@ -1,6 +1,11 @@
+import com.vanniktech.maven.publish.JavadocJar.Dokka
+import com.vanniktech.maven.publish.MavenPublishBaseExtension
+import com.vanniktech.maven.publish.KotlinMultiplatform
+
 plugins {
   kotlin("jvm")
   `java-library`
```
Comment (swankjesse) [inline tempest-testing/build.gradle.kts:7]: Same

File: tempest/build.gradle.kts [hunk 8]
```diff
@@ -1,6 +1,12 @@
+import com.vanniktech.maven.publish.JavadocJar.Dokka
+import com.vanniktech.maven.publish.MavenPublishBaseExtension
+import com.vanniktech.maven.publish.KotlinMultiplatform
+
+
 plugins {
   kotlin("jvm")
   `java-library`
```
Comment (swankjesse) [inline tempest/build.gradle.kts:8]: Same

File: tempest2-testing-docker/build.gradle.kts [hunk 9]
```diff
@@ -1,6 +1,11 @@
+import com.vanniktech.maven.publish.JavadocJar.Dokka
+import com.vanniktech.maven.publish.MavenPublishBaseExtension
+import com.vanniktech.maven.publish.KotlinMultiplatform
+
 plugins {
   kotlin("jvm")
   `java-library`
```
Comment (swankjesse) [inline tempest2-testing-docker/build.gradle.kts:7]: Same

File: tempest2-testing-internal/build.gradle.kts [hunk 10]
```diff
@@ -1,6 +1,11 @@
+import com.vanniktech.maven.publish.JavadocJar.Dokka
+import com.vanniktech.maven.publish.MavenPublishBaseExtension
+import com.vanniktech.maven.publish.KotlinMultiplatform
+
 plugins {
   kotlin("jvm")
   `java-library`
```
Comment (swankjesse) [inline tempest2-testing-internal/build.gradle.kts:7]: Same

File: tempest2-testing-junit4/build.gradle.kts [hunk 11]
```diff
@@ -1,6 +1,11 @@
+import com.vanniktech.maven.publish.JavadocJar.Dokka
+import com.vanniktech.maven.publish.MavenPublishBaseExtension
+import com.vanniktech.maven.publish.KotlinMultiplatform
+
 plugins {
   kotlin("jvm")
   `java-library`
```
Comment (swankjesse) [inline tempest2-testing-junit4/build.gradle.kts:7]: Same

File: tempest2-testing-junit5/build.gradle.kts [hunk 12]
```diff
@@ -1,6 +1,11 @@
+import com.vanniktech.maven.publish.JavadocJar.Dokka
+import com.vanniktech.maven.publish.MavenPublishBaseExtension
+import com.vanniktech.maven.publish.KotlinMultiplatform
+
 plugins {
   kotlin("jvm")
   `java-library`
```
Comment (swankjesse) [inline tempest2-testing-junit5/build.gradle.kts:7]: Same

File: tempest2-testing-jvm/build.gradle.kts [hunk 13]
```diff
@@ -1,6 +1,11 @@
+import com.vanniktech.maven.publish.JavadocJar.Dokka
+import com.vanniktech.maven.publish.MavenPublishBaseExtension
+import com.vanniktech.maven.publish.KotlinMultiplatform
+
 plugins {
   kotlin("jvm")
   `java-library`
```
Comment (swankjesse) [inline tempest2-testing-jvm/build.gradle.kts:7]: Same

File: tempest2-testing/build.gradle.kts [hunk 14]
```diff
@@ -1,6 +1,11 @@
+import com.vanniktech.maven.publish.JavadocJar.Dokka
+import com.vanniktech.maven.publish.MavenPublishBaseExtension
+import com.vanniktech.maven.publish.KotlinMultiplatform
+
 plugins {
   kotlin("jvm")
   `java-library`
```
Comment (swankjesse) [inline tempest2-testing/build.gradle.kts:7]:
```suggestion
```

File: tempest2-testing-jvm/build.gradle.kts (hunk 13 above)
Comment (swankjesse) [reply tempest2-testing-jvm/build.gradle.kts:7]:
```suggestion
```

File: tempest2-testing-junit5/build.gradle.kts (hunk 12 above)
Comment (swankjesse) [reply tempest2-testing-junit5/build.gradle.kts:7]:
```suggestion
```

File: tempest2-testing-junit4/build.gradle.kts (hunk 11 above)
Comment (swankjesse) [reply tempest2-testing-junit4/build.gradle.kts:7]:
```suggestion
```

File: tempest2-testing-internal/build.gradle.kts (hunk 10 above)
Comment (swankjesse) [reply tempest2-testing-internal/build.gradle.kts:7]:
```suggestion
```

File: tempest2-testing-docker/build.gradle.kts (hunk 9 above)
Comment (swankjesse) [reply tempest2-testing-docker/build.gradle.kts:7]:
```suggestion
```

File: tempest-internal/build.gradle.kts (hunk 1 above)
Comment (swankjesse) [inline tempest-internal/build.gradle.kts:7]:
```suggestion
```

File: tempest-testing-docker/build.gradle.kts (hunk 2 above)
Comment (swankjesse) [inline tempest-testing-docker/build.gradle.kts:7]:
```suggestion
```

File: tempest-testing-internal/build.gradle.kts (hunk 3 above)
Comment (swankjesse) [inline tempest-testing-internal/build.gradle.kts:7]:
```suggestion

```

File: tempest-testing-junit5/build.gradle.kts (hunk 5 above)
Comment (swankjesse) [inline tempest-testing-junit5/build.gradle.kts:7]:
```suggestion
```

File: tempest-testing/build.gradle.kts (hunk 7 above)
Comment (swankjesse) [inline tempest-testing/build.gradle.kts:7]:
```suggestion
```

File: tempest/build.gradle.kts (hunk 8 above)
Comment (swankjesse) [inline tempest/build.gradle.kts:8]:
```suggestion
```

File: tempest2-testing-docker/build.gradle.kts (hunk 9 above)
Comment (swankjesse) [inline tempest2-testing-docker/build.gradle.kts:7]:
```suggestion
```

File: tempest2-testing-internal/build.gradle.kts (hunk 10 above)
Comment (swankjesse) [inline tempest2-testing-internal/build.gradle.kts:7]:
```suggestion
```

File: tempest2-testing-junit4/build.gradle.kts (hunk 11 above)
Comment (swankjesse) [inline tempest2-testing-junit4/build.gradle.kts:7]:
```suggestion
```

File: tempest2-testing-junit5/build.gradle.kts (hunk 12 above)
Comment (swankjesse) [inline tempest2-testing-junit5/build.gradle.kts:7]:
```suggestion
```

File: tempest2-testing-jvm/build.gradle.kts (hunk 13 above)
Comment (swankjesse) [inline tempest2-testing-jvm/build.gradle.kts:7]:
```suggestion
```

File: tempest2-testing/build.gradle.kts (hunk 14 above)
Comment (swankjesse) [inline tempest2-testing/build.gradle.kts:7]:
```suggestion
```

File: tempest2/build.gradle.kts [hunk 15]
```diff
@@ -1,6 +1,11 @@
+import com.vanniktech.maven.publish.JavadocJar.Dokka
+import com.vanniktech.maven.publish.MavenPublishBaseExtension
+import com.vanniktech.maven.publish.KotlinMultiplatform
+
 plugins {
   kotlin("jvm")
   `java-library`
```
Comment (swankjesse) [inline tempest2/build.gradle.kts:7]:
```suggestion
```

**Review by swankjesse — COMMENTED** [review]
I think you can delete all the entries in gradle.properties that start like this:

```
POM_
```

File: build.gradle.kts [hunk 16]
```diff
@@ -18,6 +20,8 @@ buildscript {
     }
 }
 
+apply(plugin = "com.vanniktech.maven.publish.base")
+
 subprojects {
```
Comment (swankjesse) [inline build.gradle.kts:25]: Might need this?

```suggestion
allprojects {
  group = project.property("GROUP") as String
  version = project.property("VERSION_NAME") as String
}

subprojects {
```

**Review by swankjesse — COMMENTED** [review]
Delete all the `gradle.properties` files that are in submodules? They’re not necessary as long as their POM_ARTIFACT_ID matches the directory name.

**Review by swankjesse — APPROVED** [review]

File: tempest-internal/build.gradle.kts (hunk 1 above)
Comment (kyeotic) [reply tempest-internal/build.gradle.kts:7]: The other deletions where a find/replace mistake. According to [gradle docs](https://docs.gradle.org/current/userguide/java_library_plugin.html) this is necessary for the `api/implementation` functions.

**Review by swankjesse — COMMENTED** [review]
LGTM!

**Pull request merged**